common/src/main/java/com/wynntils/features/user/ChatMentionFeature.java

```diff
@@ -0,0 +1,116 @@
+/*
+ * Copyright © Wynntils 2023.
+ * This file is released under AGPLv3. See LICENSE for full license details.
+ */
+package com.wynntils.features.user;
+
+import com.wynntils.core.config.Config;
+import com.wynntils.core.config.ConfigHolder;
+import com.wynntils.core.features.UserFeature;
+import com.wynntils.handlers.chat.event.ChatMessageReceivedEvent;
+import com.wynntils.utils.mc.ComponentUtils;
+import com.wynntils.utils.mc.McUtils;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import net.minecraft.network.chat.Component;
+import net.minecraft.network.chat.MutableComponent;
+import net.minecraft.sounds.SoundEvents;
+import net.minecraftforge.eventbus.api.SubscribeEvent;
+
+public class ChatMentionFeature extends UserFeature {
+    @Config
+    public String aliases = "";
+
+    private Pattern pattern = Pattern.compile(
+            "(?<!\\[)\\b(" + McUtils.mc().getUser().getName()
+                    + (aliases.length() > 0 ? "|" + aliases.replace(",", "|") : "") + ")\\b(?!:|])",
+            Pattern.CASE_INSENSITIVE);
+
+    @Override
+    protected void onConfigUpdate(ConfigHolder configHolder) {
+        pattern = Pattern.compile(
+                "(?<!\\[)\\b(" + McUtils.mc().getUser().getName()
+                        + (aliases.length() > 0 ? "|" + aliases.replace(",", "|") : "") + ")\\b(?!:|])",
+                Pattern.CASE_INSENSITIVE);
+    }
+
+    @SubscribeEvent()
+    public void onChat(ChatMessageReceivedEvent e) {
+        Component message = e.getMessage();
+
+        Matcher looseMatcher = pattern.matcher(ComponentUtils.getUnformatted(message));
```

Comment:
This all looks very complicated. Can't you just convert it to a coded string and look for a regexp in that? That's `ComponentUtils.getCoded()` instead. And then you do not need to match twice.

Reply:
The first match is to check if the message should be processed at all. Then the components are modified in recurseMark in a way which should preserve the raw components as well as possible in order to not edit what doesn't need to be edited.

Reply:
Ah, I missed the fact that you actually *rewrite* the chat message, and not only look for a pattern to play a sound. Ok, then I realize why you do this. 

Some suggestions:
1) These are two different behaviors -- pling on your name mentioned, and rewriting the chat to color your name. Please add config options for these two. They can be true by default, but users should be able to opt-out of that which they might not want.

2) The "recurseMark" is not a very descriptive name. Please name the method after what it is trying to achieve, not after how it is implemented. In this case, perhaps "rewriteComponentWithHighlight", or something like that. (Not the best of names, if you can get the message across with a more succinct name, feel free to use that instead).

3) Add some comment at the place where you actually rewrite the message, to highlight the fact that the message is changed at that point. All the rest is just boilerplate for processing the Component tree.

4) Maybe the user should be able to select what color code the highlight use?

Reply:
Alright I've implemented all of these but the fourth. I'm not sure how changing the color could be done nicely as the highlight has to be styled with §e rather than using propper styling to not break guild chat.

Reply:
```
@Config
ChatFormatting rewriteColorCode;
```